scalameter-core/src/main/scala/org/scalameter/execution/invocation/InvocationCountMatcher.scala

```diff
@@ -0,0 +1,104 @@
+package org.scalameter.execution.invocation
+
+import java.lang.reflect.Method
+import java.util.regex.Pattern
+import org.objectweb.asm.Type
+import scala.util.matching.Regex
+import InvocationCountMatcher._
+
+
+/** Object that matches the methods whose invocations should be counted.
+ *
+ *  @param classMatcher matches full class name given in internal format ('/' instead of a '.' as a separator)
+ *  @param methodMatcher matches a specific method
+ */
+case class InvocationCountMatcher(classMatcher: ClassMatcher, methodMatcher: MethodMatcher) {
+  def classMatches(className: String): Boolean = classMatcher.matches(className)
+
+  def methodMatches(methodName: String, methodDescriptor: String): Boolean = methodMatcher.matches(methodName, methodDescriptor)
+}
+
+object InvocationCountMatcher {
+  /** Mixin used for selecting classes whose methods will be checked against [[MethodMatcher]] and if matched,
+   *  counted by a method invocation counting measurer.
+   */
+  sealed trait ClassMatcher {
+    /** Matches class name given either in a normal format or internal jvm format ('/' instead of a '.').
```

Comment:
That makes sense, but in that case your instrumentation code should do the transformation internally. That way you avoid duplicating code across `matches` implementations, and you simplify the contract in case we at some point allow clients to define their own matchers (or add new matchers, e.g. composite matchers like `Not`, `And` and `Or`, ourselves).